crypto.sh corrupts .passed/.failed counters on Solaris and all other
systems, which compile tcpdump without crypto support. The bug has
been reported to upstream as #648 [1]

[1] https://github.com/the-tcpdump-group/tcpdump/issues/648

--- a/tests/crypto.sh	2017-09-04 01:17:14.000000000 +0200
+++ b/tests/crypto.sh	2017-11-15 15:19:59.351099113 +0100
@@ -4,10 +4,10 @@
 
 # Only attempt OpenSSL-specific tests when compiled with the library.
 
+passed=`cat .passed`
+failed=`cat .failed`
 if grep '^#define HAVE_LIBCRYPTO 1$' ../config.h >/dev/null
 then
-	passed=`cat .passed`
-	failed=`cat .failed`
 	if ./TESTonce esp1 02-sunrise-sunset-esp.pcap esp1.out '-E "0x12345678@192.1.2.45 3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758"'
 	then
 		passed=`expr $passed + 1`
